pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_drop_constraint.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package scbuildstmt
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scerrors"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
+	"github.com/cockroachdb/errors"
+)
+
+func alterTableDropConstraint(
+	b BuildCtx, tn *tree.TableName, tbl *scpb.Table, t *tree.AlterTableDropConstraint,
+) {
+	constraintElems := b.ResolveConstraint(tbl.TableID, t.Constraint, ResolveParams{
+		IsExistenceOptional: t.IfExists,
+		RequiredPrivilege:   privilege.CREATE,
+	})
+	if constraintElems == nil {
+		// constraint not found but `IF EXISTS` is set
```

Comment:
nit: make the comment a sentence.

While you're here, can we buffer a notice like postgres does?

```
postgres=# create table foo (i int primary key);
CREATE TABLE
postgres=# create table bar (i int primary key, j int references foo(i));
CREATE TABLE
postgres=# \d bar
                Table "public.bar"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 i      | integer |           | not null | 
 j      | integer |           |          | 
Indexes:
    "bar_pkey" PRIMARY KEY, btree (i)
Foreign-key constraints:
    "bar_j_fkey" FOREIGN KEY (j) REFERENCES foo(i)

postgres=# alter table bar drop constraint if exists bar_j_fkey;
ALTER TABLE
postgres=# alter table bar drop constraint if exists bar_j_fkey;
NOTICE:  constraint "bar_j_fkey" of relation "bar" does not exist, skipping
ALTER TABLE
```

Reply:
If it ends up being painful, can you file an issue to buffer such a notice?

Reply:
no problem, this is rather easy. Added!